app/src/main/java/org/sil/storyproducer/model/StoryIO.kt

```diff
@@ -39,41 +43,54 @@ fun storyFromJson(context: Context, storyTitle: String): Story?{
                 ?: return null
         return adapter.fromJson(fileContents)
     } catch (e: Exception) {
-        return null
+        throw StoryIOException("StoryIO: storyFromJson() parse exception from moshi")
     }
 }
 
-fun parseStoryIfPresent(context: Context, storyPath: androidx.documentfile.provider.DocumentFile): Story? {
```

Comment:
@jakedallinson I revised #555 list of what To Do - please review that. It may be that we don't need to change much if anything in parseStoryIfPresent(). But many of your comments would be helpful to keep! 
I am thinking I would like to implement #555 in v3.0 beta/release, so for that objective we need to minimize any logic changes and thereby reduce the risk of introducing a new bug into our release. :-(

I think the original code in parseStoryIfPresent() is intending to perform this: 'if there's no existing project (story.json), then this must be a new project, so initialize the project by importing the BloomHTML'. 
However, the code ineptly handles the json deserialization error the same as if story.json does not exist, and proceeds to the BloomHTML.
Anyway, at this time (for v3.0) I now think it's okay for the deserialization error scenario to return a blank story from the BloomHTML.